### PR TITLE
JS & CSS optimizations

### LINKED ISSUE #125: Compress and CSS and Javascript
Add options to node-sass to compress the output CSS, and add something to the build script to compress the javascript (or configure cloudflare)

### LINKED ISSUE #126: Clean out what script tags we are using
- Remove unnecessary script tags (do we need popper.js and jquery?)
- index.js has two script tags
- make sure necessary CDN loaded scripts use a good CDN with a long cache time
- add `defer` to basically all script tags as no javascript should be necessary to do a "first paint" of the website

### PR DESCRIPTION
**Pull request by phulsechinmay**
Closes #125 
* Production build now creates compressed CSS
* Cloudflare is configured to minify JS

Closes #126 
* Removed unnecessary script tags
* Added defer to all script tags

### CODE REVIEW

**Comment by geooot** [comment]
@phulsechinmay 
Apparently the performance got worse according to lighthouse (using webpagetest.org/lighthouse).

Before:
https://www.webpagetest.org/result/200511_J9_56441e9ce90f67148686c57acbe245c1/
image

After:
https://www.webpagetest.org/result/200511_KS_26acd0d1b05b5749d02cb794adb7669f/#performance
image

I think it might be that we have to use `defer="async"`
Also we should take advantage of this setting
image

**Comment by phulsechinmay** [comment]
@geooot Was the original test on the base `tamudatathon.com` website? There might be cloudflare caching that might improving performance. 

Will make the switch to async and the preconnect additions

**Comment by geooot** [comment]
@phulsechinmay Just noticed that, here is the test based off of gigabowser.now.sh https://www.webpagetest.org/result/200511_W5_b87bee4838763d63d1b00eee4694e6cd/

**Comment by phulsechinmay** [comment]
image
@geooot This is the latest performance with defer tags removed. I think defer tags are causing problems with our 'Time To Interactive' score, so might be a good idea to leave them out.

**Comment by geooot** [comment]
@phulsechinmay Looks like it, could we try adding `async="async"` to all script tags in the head

File: views/partials/head.ejs [hunk 1]
```diff
@@ -13,19 +13,24 @@
   <meta name="twitter:title" content="<%= req.pageName %> - TAMU Datathon">
   <meta name="twitter:description" content="">
 
+  <!-- Specify preconnect hosts -->
+  <link rel="preconnect" href="https://unpkg.com" />
+  <link rel="preconnect" href="https://kit.fontawesome.com" />
+  <link rel="preconnect" href="https://www.googletagmanager.com" />
+  <link rel="preconnect" href="https://www.google-analytics.com" />
```
Comment (geooot) [inline views/partials/head.ejs:20]: I think this has to be:
```suggestion
  <link rel="preconnect" href="https://www.googletagmanager.com" />
```

Comment (phulsechinmay) [reply views/partials/head.ejs:20]: I think the google tag manager makes a call to google-analytics. The lighthouse report specifies `google-analytics.com` when the preconnect for it isn't there

**Comment by phulsechinmay** [comment]
image
@geooot Performance with async in head script tags. 3 points lower than when none of them have async

**Comment by phulsechinmay** [comment]
image
@geooot Latest test. Anything else we can improve on?

**Comment by geooot** [comment]
@phulsechinmay Nah I think this is fine, I'm going to merge it

**Comment by phulsechinmay** [comment]
image

@geooot Lighthouse still wants us to add the google analytics preconnect

**Pull request merged**